### ISSUE TITLE
Outdated pg_dump binary

### ISSUE THREAD

**Issue by Cludch** [b3vis/docker-borgmatic#59]
Hey,

I just tried to dump a PostgreSQL server 13 database.
However, the `pg_dump` binary installed within the latest image is incompatible with PostgreSQL 13.1.
The version of `pg_dump` is 12.4.
Is there a way to update this?

The package seems to have been updated: https://pkgs.alpinelinux.org/package/edge/main/x86/postgresql-client
So maybe a rebuild would suffice?

**Comment by b3vis** [comment]
Hi, 

I didn't want to upgrade the base image to use the edge repositories from Alpine as it might introduce some breaking changes. I have however created a develop branch for you which does use the edge repositories. I will setup a automated build on Docker Hub for you so you can pull this down as b3vis/borgmatic:develop

```
λ docker run -it --rm b3vis/borgmatic:develop pg_dump --version
pg_dump (PostgreSQL) 13.1
```

**Comment by Cludch** [comment]
Oh, I didn't see that this was only updated on the edge repository.
Thanks a lot for the `develop` image! I highly appreciate your fast reply.

**Comment by b3vis** [comment]
Any feedback @Cludch?

**Comment by Cludch** [comment]
Oh, sorry for the unclear response!
The `develop` image works great. Thanks a lot!

**Comment by 917huB** [comment]
I noticed this update in the borgmatic repo too. 
https://projects.torsion.org/witten/borgmatic/commit/4897a78fd3c90e801dc9bcd0d9b3ae2e539178e8

**Comment by Bogey** [comment]
:develop is working for me too! 👍🏻 

Question: does the develop also include ntfy?

**Comment by jejbq** [comment]
Hello,

pg_dump 16.1 is available in alpine 3.19.

So I suggest to rebase the Dockerfile to 3.11-alpine3.19 (from python:3.11.5-alpine3.18 currently)
or better directly use python:3.11-alpine or python:3-alpine

https://hub.docker.com/_/python
3.11.7-alpine3.19, 3.11-alpine3.19, 3.11.7-alpine, 3.11-alpine
-or-
3.12.1-alpine3.19, 3.12-alpine3.19, 3-alpine3.19, alpine3.19, 3.12.1-alpine, 3.12-alpine, 3-alpine, alpine

``` patch
diff --git a/Dockerfile b/Dockerfile
index 39bf473..8717dcc 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # syntax = docker/dockerfile:latest
-FROM python:3.11.5-alpine3.18
+FROM python:3.11-alpine
 LABEL maintainer='github.com/borgmatic-collective'
 VOLUME /mnt/source
 VOLUME /mnt/borg-repository
@@ -22,12 +22,13 @@ RUN apk add --update --no-cache \
     mariadb-client \
     mariadb-connector-c \
     mongodb-tools \
-    openssl1.1-compat \
+    openssl \
     postgresql-client \
     sqlite \
     sshfs \
     supercronic \
     tzdata \
+    && apk upgrade --update --no-cache \
     && rm -rf \
     /var/cache/apk/* \
     /.cache
```
